### PR TITLE
Rotate dendrograms

### PR DESCRIPTION
**Pull request by whelena**
# Description
Support rotating dendrogram trees

## Small example
Screenshot 2025-04-14 at 22 01 19

# Checklist

- [ ] This PR **does *NOT* contain** Protected Health Information [(PHI)](https://ohrpp.research.ucla.edu/hipaa/). A repo may ***need to be deleted*** if such data is uploaded. <br> Disclosing PHI is a ***major problem***[^1] - Even ***a small leak can be costly***[^2].
  
- [ ] This PR **does *NOT* contain** germline genetic data[^3], RNA-Seq, DNA methylation, microbiome or other molecular data[^4].

[^1]: [UCLA Health reaches $7.5m settlement over 2015 breach of 4.5m patient records](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m)
[^2]: [The average healthcare data breach costs $2.2 million, despite the majority of breaches releasing fewer than 500 records.](https://www.ponemon.org/local/upload/file/Sixth%20Annual%20Patient%20Privacy%20%26%20Data%20Security%20Report%20FINAL%206.pdf)
[^3]: [Genetic information is considered PHI.](https://www.genome.gov/about-genomics/policy-issues/Privacy#:~:text=In%202013%2C%20as%20required%20by,genetic%20information%20for%20underwriting%20purposes.)
  [Forensic assays can identify patients with as few as 21 SNPs](https://www.sciencedirect.com/science/article/pii/S1525157817305962)
[^4]: [RNA-Seq](https://www.nature.com/articles/ng.2248), [DNA methylation](https://ieeexplore.ieee.org/document/7958619), [microbiome](https://www.pnas.org/doi/pdf/10.1073/pnas.1423854112), or other molecular data can be used to predict genotypes (PHI) and reveal a patient's identity.

- [ ] This PR **does *NOT* contain** other non-plain text files, such as: compressed files, images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other output files.

_&emsp; To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example._

- [ ] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [ ] I have set up or verified the `main` branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [ ] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].
  
- [ ] I have added the major changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

### CODE REVIEW

**Comment by whelena** [comment]
To do:
- write unit tests
- test on larger dendrograms to identify edge cases
- document new functions

File: R/angles.R [hunk 1]
```diff
@@ -18,9 +18,9 @@ calculate.angles.radial <- function(
 
         parent.id <- tree$parent[tree$tip == current.node.id];
 
-        if (parent.id == -1) {
-            tree$angle[tree$tip == current.node.id] <- start.angle;
-            }
+        # if (parent.id == -1) {
```
Comment (dan-knight) [inline R/angles.R:21]: Are you sure don't need this root node case anymore?

File: R/angles.R [hunk 2]
```diff
@@ -63,6 +67,7 @@ calculate.angles.radial <- function(
                     v = v,
                     dx.scale = median(tree$length1)
                     );
+                # angles[tree$tip %in% child.ids] <- start.angle + angles[tree$tip %in% child.ids];
```
Comment (dan-knight) [inline R/angles.R:70]: Same question here. We're sure that `split.equal.x.dist` accounts for the parent angle properly?

Comment (whelena) [reply R/angles.R:70]: This is from my naive try to just add `start.angle` to the dendrogram, but it doesn't exist prior to this PR.

File: tests/testthat/test-rotate.R [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    df <- data.frame(
+        basex = c(1, 2),
+        basey = c(0, 0),
+        tipx = c(1, 2),
+        tipy = c(1, 1)
+        );
+
+    rotated <- rotate.dendrogram(df, rotate.by = pi / 2);
+
+    expect_equal(dim(rotated), dim(df));
```
Comment (dan-knight) [inline tests/testthat/test-rotate.R:35]: Fine for now. If we start getting more complex failures, it may be good to split each requirement into separate unit test cases. As it's written now, if theres a failure on line 39, line 40 won't be checked (for example).

**Review by dan-knight — APPROVED** [review]
